src/Simplex/Chat/Store/Messages.hs

```diff
@@ -1106,34 +1084,24 @@ getDirectChatAround_ db user ct@Contact {contactId} aroundItemId count search =
   beforeChatItems <- liftIO $ reverse <$> mapM (safeGetDirectItem db user ct currentTs) beforeIds
   afterChatItems <- liftIO $ mapM (safeGetDirectItem db user ct currentTs) afterIds
   let chatItems = beforeChatItems <> [middleChatItem] <> afterChatItems
-  gapToLatest <- liftIO $ getDirectChatItemGapToLatest_ db user ct (last chatItems) search
-  let chatGap = if gapToLatest > 0 then Just gapToLatest else Nothing
-  pure (Chat (DirectChat ct) chatItems stats, chatGap)
+  pure $ Chat (DirectChat ct) chatItems stats
 
-getDirectChatInitial_ :: DB.Connection -> User -> Contact -> Int -> ExceptT StoreError IO (Chat 'CTDirect, Maybe ChatGap)
+getDirectChatInitial_ :: DB.Connection -> User -> Contact -> Int -> ExceptT StoreError IO (Chat 'CTDirect)
 getDirectChatInitial_ db user@User {userId} ct@Contact {contactId} count = do
   firstUnreadItemId_ <- liftIO getDirectChatMinUnreadItemId_
   case firstUnreadItemId_ of
     Just firstUnreadItemId -> do
-      (chat, gap) <- getDirectChatAround_ db user ct firstUnreadItemId count ""
-      case gap of
-        Just size -> do
-          if size > 0
-            then getLatestItems_ chat size
-            else pure (chat, Nothing)
-        Nothing -> pure (chat, Nothing)
-    Nothing -> liftIO $ (,Nothing) <$> getDirectChatLast_ db user ct count ""
+      chat <- getDirectChatAround_ db user ct firstUnreadItemId count ""
+      let items = chatItems chat
+      if null items || length items == count
+        then pure chat
+        else do
+          let remainingCount = count - length items
```

Comment:
maybe move remainingCount to outside so that it guarantess negative count doesn't get into LIMIT
```suggestion
      let items = chatItems chat
           remainingCount = count - length items
      if null items || remainingCount <= 0
        then pure chat
        else do
```